File: api/schemas/export.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field, field_validator


class ExportRequest(BaseModel):
    finding_ids: list[str] = Field(min_length=1)

    @field_validator("finding_ids")
    @classmethod
    def strip_and_deduplicate(cls, value: list[str]) -> list[str]:
        seen: set[str] = set()
        result: list[str] = []
        for item in value:
            stripped = item.strip()
            if not stripped:
                raise ValueError("finding_ids must not contain blank entries")
            if stripped not in seen:
                seen.add(stripped)
                result.append(stripped)
        return result


class ExportEvidenceItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    evidence_type: str
    fact_id: str | None = None
    calculation_key: str | None = None
    reference_key: str | None = None
    citation_label: str | None = None
    citation_url: str | None = None
    excerpt: str | None = None


class ExportFindingItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    finding_id: str
    finding_key: str
    title: str
    summary: str
    severity: str
    claim_type: str
    uncertainty_state: str
    confidence: float | None = None
    evidence: list[ExportEvidenceItem] = Field(default_factory=list)


class ExportRejectedItem(BaseModel):
    finding_id: str
    reason: str


class ExportSummary(BaseModel):
    case_id: str
    analysis_run_id: str
    exported_at: datetime
    finding_count: int
    findings: list[ExportFindingItem]
    rejected: list[ExportRejectedItem] = Field(default_factory=list)

File: src/api/export.ts
import { endpoint, responseError } from './client';

export interface ExportEvidenceItem {
  evidence_type: string;
  fact_id: string | null;
  calculation_key: string | null;
  reference_key: string | null;
  citation_label: string | null;
  citation_url: string | null;
  excerpt: string | null;
}

export interface ExportFindingItem {
  finding_id: string;
  finding_key: string;
  title: string;
  summary: string;
  severity: string;
  claim_type: string;
  uncertainty_state: string;
  confidence: number | null;
  evidence: ExportEvidenceItem[];
}

export interface ExportRejectedItem {
  finding_id: string;
  reason: string;
}

export interface ExportSummary {
  case_id: string;
  analysis_run_id: string;
  exported_at: string;
  finding_count: number;
  findings: ExportFindingItem[];
  rejected: ExportRejectedItem[];
}

export interface DraftSection {
  heading: string;
  body: string;
}

export interface DraftResult {
  case_id: string;
  analysis_run_id: string;
  generated_at: string;
  sections: DraftSection[];
  filtered_phrases: string[];
  warnings: string[];
}

export async function exportFindings(
  caseId: string,
  findingIds: string[],
): Promise<ExportSummary> {
  const response = await fetch(endpoint(`/api/cases/${caseId}/export`), {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ finding_ids: findingIds }),
  });
  if (!response.ok) {
    throw await responseError(response, 'No pudimos exportar los hallazgos.');
  }
  return response.json();
}

export async function generateDraft(
  caseId: string,
  findingIds: string[],
): Promise<DraftResult> {
  const response = await fetch(endpoint(`/api/cases/${caseId}/draft`), {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ finding_ids: findingIds }),
  });
  if (!response.ok) {
    throw await responseError(response, 'No pudimos generar el borrador.');
  }
  return response.json();
}
